digits/dataset/images/classification/views.py

```diff
@@ -314,18 +318,19 @@ def image_classification_dataset_create():
         if request_wants_json():
             return flask.jsonify(job.json_dict())
         else:
-            return flask.redirect(flask.url_for('datasets_show', job_id=job.id()))
+            return flask.redirect(flask.url_for('datasets_show', job_id=job.id())+'?workspace='+workspace)
 
     except:
         if job:
             scheduler.delete_job(job)
         raise
 
-def show(job):
+def show(job, *args):
```

Comment:
Maybe call the second parameter `workspace` rather than `args`, because that's the only way it's called.